intdataplane/int_dataplane.go

```diff
@@ -455,6 +463,67 @@ func (d *InternalDataplane) loopUpdatingDataplane() {
 	}
 }
 
+func (d *InternalDataplane) configureKernel() {
+	// For IPv4, we rely on the kernel's reverse path filtering to prevent workloads from
+	// spoofing their IP addresses.
+	//
+	// The RPF check for a particular interface is controlled by several sysctls:
+	//
+	//     - ipv4.conf.all.rp_filter is a global override
+	//     - ipv4.conf.default.rp_filter controls the value that is set on a newly created
+	//       interface
+	//     - ipv4.conf.<interface>.rp_filter controls a particular interface.
+	//
+	// The algorithm for combining the global override and per-interface values is to take the
+	// *numeric* maximum between the two.  The values are: 0=off, 1=strict, 2=loose.  "loose"
+	// is not suitable for Calico since it would allow workloads to spoof packets from other
+	// workloads on the same host.  Hence, we need the global override to be <=1 or it would
+	// override the per-interface setting to "strict" that we require.
+	//
+	// Unless the IgnoreLooseRPF flag is set, we bail out rather than simply setting it
+	// because setting 2, "loose", is unusual and it is likely to have been set deliberately.
+	rpFilter, err := readRPFilter()
+	if err != nil {
+		logCxt := log.WithError(err)
+		if d.config.IgnoreLooseRPF {
+			logCxt.Error("Failed to read kernel's rp_filter value from /proc/sys. " +
+				"Ignoring due to IgnoreLooseRPF setting.")
+		} else {
+			logCxt.Fatal("Failed to read kernel's rp_filter value from /proc/sys")
+		}
+	} else if rpFilter > 1 {
+		if d.config.IgnoreLooseRPF {
+			log.Warn("Kernel's RPF check is set to 'loose' and IgnoreLooseRPF " +
+				"set to true.  Calico will not be able to prevent workloads " +
+				"from spoofing their source IP.  Please ensure that some " +
+				"other anti-spoofing mechanism is in place (such as running " +
+				"only non-privileged containers).")
+		} else {
+			log.Fatal("Kernel's RPF check is set to 'loose'.  This would " +
+				"allow endpoints to spoof their IP address.  Calico " +
+				"requires net.ipv4.conf.all.rp_filter to be set to " +
+				"0 or 1.")
```

Comment:
Perhaps add: "or if you really need the global loose RPF setting, configure IgnoreLooseRPF to tell Calico to allow this - but be aware that then ..."